### PR TITLE
RF: enable oneDAL predict proba

### PR DESCRIPTION
**Pull request by agorshk**

### CODE REVIEW

File: daal4py/sklearn/ensemble/_forest.py [hunk 1]
```diff
@@ -643,24 +643,20 @@ def predict_proba(self, X):
             The class probabilities of the input samples. The order of the
             classes corresponds to that in the attribute :term:`classes_`.
         """
-        # Temporary solution
         X = check_array(X, accept_sparse=['csr', 'csc', 'coo'])
```
Comment (PetrovKP) [inline daal4py/sklearn/ensemble/_forest.py:646]:
```suggestion
        X = check_array(X, accept_sparse=['csr', 'csc', 'coo'], dtype=[
                np.float64, np.float32])
```

File: daal4py/sklearn/ensemble/_forest.py [hunk 2]
```diff
@@ -643,24 +643,20 @@ def predict_proba(self, X):
             The class probabilities of the input samples. The order of the
             classes corresponds to that in the attribute :term:`classes_`.
         """
-        # Temporary solution
         X = check_array(X, accept_sparse=['csr', 'csc', 'coo'])
+
+        if not hasattr(self, 'daal_model_') or \
+                sp.issparse(X) or self.n_outputs_ != 1 or \
+                not (X.dtype == np.float64 or X.dtype == np.float32) or \
```
Comment (PetrovKP) [inline daal4py/sklearn/ensemble/_forest.py:650]:
```suggestion
                

```

File: daal4py/sklearn/ensemble/_forest.py (hunk 1 above)
Comment (agorshk) [reply daal4py/sklearn/ensemble/_forest.py]: thanks, it was fixed

File: daal4py/sklearn/ensemble/_forest.py (hunk 2 above)
Comment (agorshk) [reply daal4py/sklearn/ensemble/_forest.py]: thanks, it was fixed

**Review by PetrovKP — APPROVED** [review]

**Pull request merged**